pkg/operation/botanist/controlplane/kubecontrollermanager/waiter.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubecontrollermanager
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/operation/common"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/retry"
+)
+
+func (k *kubeControllerManager) Wait(_ context.Context) error        { return nil }
+func (k *kubeControllerManager) WaitCleanup(_ context.Context) error { return nil }
+
+func (k *kubeControllerManager) WaitForControllersToBeActive(ctx context.Context) error {
+	type controllerInfo struct {
+		name   string
+		labels map[string]string
+	}
+
+	type checkOutput struct {
```

Comment:
Can this coding related to parallelism be removed now? There is only one KCM pod for a cluster, so there is no need to parallelise anything.